lamindb_setup/dev/_hub_crud.py

```diff
@@ -1,6 +1,26 @@
 from postgrest.exceptions import APIError
 from supabase.client import Client
+from supafunc.errors import FunctionsRelayError, FunctionsHttpError
 from typing import Optional, Dict
+import lamindb_setup
+from lamin_utils import logger
+import os
+
+
+def access_aws(client: Client):
+    response = None
+    try:
+        response = client.functions.invoke("access-aws")
+    except (FunctionsRelayError, FunctionsHttpError) as exception:
+        err = exception.to_dict()
+        logger.warning(err.get("message"))
+    if response is not None:
+        credentials = response["Credentials"]
+        os.environ["AWS_ACCESS_KEY_ID"] = credentials["AccessKeyId"]
+        os.environ["AWS_SECRET_ACCESS_KEY"] = credentials["SecretAccessKey"]
+        os.environ["AWS_SESSION_TOKEN"] = credentials["SessionToken"]
```

Comment:
As we discussed, it is better to pass the credentials to fsspec here https://github.com/laminlabs/lamindb-setup/blob/8a55c63fc42fdb631c6e2ecf6414e474865aedae/lamindb_setup/dev/_settings_storage.py#L69

Reply:
Yes and thanks for linking the code, Sergei! This is my very first draft to play around with the experience. We'll do as you say.

Reply:
One thing we also have to do is to introduce a fallback mechanism that makes it possible to access public buckets via `anon` in case the credentials don't have the policy for it.

Reply:
https://laminlabs.slack.com/archives/C04TMJPNCBS/p1706049778482779

Reply:
Ah, ok, I see. I can work on passing the credentials to s3fs and on the fallback when we can get them successfully from the hub.

But also would be geat to avoid the fallback if possible. I will check the  slack thread later.